wpiutil/src/main/java/edu/wpi/first/util/CircularBuffer.java

```diff
@@ -103,14 +103,14 @@ public void addLast(double value) {
    * Pop value at front of buffer.
    *
    * @return value at front of buffer
+   * @throws NoSuchElementException if the buffer is empty
```

Comment:
Maybe add `*OrNull()`/`*OrDefault()` variants to all of these throwing functions?